libs/langchain/langchain/vectorstores/matching_engine.py

```diff
@@ -232,7 +232,7 @@ def similarity_search_by_vector_with_score(
         filter = filter or []
 
         # If the endpoint is public we use the find_neighbors function.
-        if self.endpoint._public_match_client:
+        if hasattr(self.endpoint, "_public_match_client"):
```

Comment:
this is checking for different things, i think we still want want to check that this is True or non Nully right?